OpenRA.Game/Map/Map.cs

```diff
@@ -838,6 +840,14 @@ bool ContainsAllProjectedCellsCovering(MPos uv)
 			if (Grid.MaximumTerrainHeight == 0)
 				return Bounds.Contains(uv.U, uv.V);
 
+			// PERF: Most cells lie within a region where no matter their height,
```

Comment:
This looks like it would be a nice perf win, worth splitting into its own PR to be merged ASAP?

Reply:
#19691